src/ControlSystem/ControlErrors/Size/Info.cpp

```diff
@@ -10,6 +10,42 @@
 #include "Utilities/Serialization/CharmPupable.hpp"
 
 namespace control_system::size {
+Info::Info(const std::unique_ptr<State>& in_state, double in_damping_time,
+           double in_target_char_speed, double in_target_drift_velocity,
+           double in_suggested_time_scale,
+           bool in_discontinuous_change_has_occurred)
+    : state(in_state->get_clone()),
+      damping_time(in_damping_time),
+      target_char_speed(in_target_char_speed),
+      target_drift_velocity(in_target_drift_velocity),
+      suggested_time_scale(in_suggested_time_scale),
+      discontinuous_change_has_occurred(in_discontinuous_change_has_occurred) {}
+
+// NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init,-warnings-as-errors)
+Info::Info(const Info& rhs) {
+  set_all_but_state(rhs);
+  state = rhs.state->get_clone();
+}
+
+Info& Info::operator=(const Info& rhs) {
+  set_all_but_state(rhs);
+  state = rhs.state->get_clone();
+  return *this;
+}
+
+// NOLINTNEXTLINE(cppcoreguidelines-pro-type-member-init,-warnings-as-errors)
+Info::Info(Info&& rhs) {
+  set_all_but_state(rhs);
+  state = std::move(rhs.state);
+  rhs.state = nullptr;
```

Comment:
How is this different from the default implementation?

Reply:
I'm not sure. I know I needed the copy constructor/assignment because of the `unique_ptr`, so I just did everything. I guess I can remove the `move` constructor and assignment because that's trivial even with a `unique_ptr`. Does that sound correct?

Reply:
Remove and replace with `= default`, yes.  Defaulting things is less error-prone, and in some cases (although I think not this one) can allow some compiler optimizations.